### PR TITLE
17727 Refactored Conversion/Alteration Flow

### PR DESCRIPTION
**Pull request by leodube-aot**
*Issue #:* /bcgov/entity#17727

*Description of changes:*
- Completed refactoring of the conversion/alteration flow

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

File: src/components/new-request/search.vue [hunk 1]
```diff
@@ -291,6 +311,12 @@ export default class Search extends Mixins(CommonMixin, NrAffiliationMixin, Sear
       this.isSupportedAmalgamation(this.getEntityTypeCd)
     ) return true
 
+    if (
+      this.isConversion &&
+      this.isNumberedCompany &&
+      this.isSupportedAlteration(this.getConversionType)
+    ) return true
+
```
Comment (leodube-aot) [inline src/components/new-request/search.vue:319]: @severinbeauvais Was this sort of how you pictured us implementing logic for each flow? A separate if statement as needed?

File: src/components/new-request/search.vue [hunk 2]
```diff
@@ -93,9 +93,23 @@
 
       <!-- Conversion (aka Alteration) flow -->
       <template v-else-if="isConversion">
-        <EntityType v-if="getSearchBusiness && isAlterable" />
         <BusinessLookupFetch />
+        <EntityType v-if="getSearchBusiness && isAlterable" cols="12" md="12" />
         <CompanyType v-if="showCompanyTypeRadioButtons" />
+
+        <template v-if="isNamedCompany">
```
Comment (severinbeauvais) [inline src/components/new-request/search.vue:100]: You probably need to add `|| isSociety` here.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-688-1g710hqa.web.app

File: src/components/new-request/search.vue [hunk 3]
```diff
@@ -93,9 +93,23 @@
...
+          <v-col cols="12" :md="showDesignation ? '8' : '12'">
+            <NameInput
+              :is-mras-search="(isXproFlow && isMrasJurisdiction && !getHasNoCorpNum)"
+              @emit-corp-num-validity="corpNumValid = $event"
+            />
+          </v-col>
+          <Designation cols="12" md="4" />
```
Comment (severinbeauvais) [inline src/components/new-request/search.vue:107]: You probably need to add `v-if="showDesignation"` here.

File: src/components/new-request/search.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/components/new-request/search.vue:319]: Yup, exactly 👍

File: src/components/new-request/search.vue [hunk 4]
```diff
@@ -271,6 +285,12 @@ export default class Search extends Mixins(CommonMixin, NrAffiliationMixin, Sear
   }
 
   get showCompanyTypeRadioButtons (): boolean {
+    if (
+      this.isConversion &&
+      !!this.getSearchBusiness &&
+      !!this.getConversionType
+    ) return true
```
Comment (severinbeauvais) [inline src/components/new-request/search.vue:292]: I think you can simply add the needed logic to line 98 and then we can eventually get rid of this getter.

**Review by severinbeauvais — COMMENTED** [review]
Looks good. Do you need to support xpro or societies?

**Comment by severinbeauvais** [comment]
Did you compare exactly with Dev Namerequest?

**Comment by leodube-aot** [comment]
> Looks good. Do you need to support xpro or societies?

I'm not 100% sure. The UXPin doesn't state anything about those being part of allowable alterations but I'll double check.

> Did you compare exactly with Dev Namerequest?

I compared it using a BEN (BC0870877), a ULC (BC0871323), and a Limited Company (BC0870316) and it matched with dev

**Comment by severinbeauvais** [comment]
> > Looks good. Do you need to support xpro or societies?
> 
> I'm not 100% sure. The UXPin doesn't state anything about those being part of allowable alterations but I'll double check.
> 
> Edit: On dev I tested with an Expro (A0002744) and it gives an error message stating the business cannot alter it's type, which matches with the refactor. Weirdly, dev will also display the Jurisdiction select field despite the error (which I think is likely incorrect), but that does not happen with the refactored version. Will need to try with a society.
> 
> > Did you compare exactly with Dev Namerequest?
> 
> I compared it using a BEN (BC0870877), a ULC (BC0871323), and a Limited Company (BC0870316) and it matched with dev

Actually, I remembered: societies can't alter their type,

I think you're right that that's a bug in Dev,

Looks good to me!

File: src/components/new-request/search.vue (hunk 2 above)
Comment (severinbeauvais) [reply src/components/new-request/search.vue:100]: Societies cannot change their type -- don't add this condition. Resolving.

File: src/components/new-request/search.vue (hunk 4 above)
Comment (severinbeauvais) [reply src/components/new-request/search.vue]: Clean! I love it!

File: src/components/new-request/search.vue [hunk 5]
```diff
@@ -93,9 +93,23 @@
...
+              :is-mras-search="(isXproFlow && isMrasJurisdiction && !getHasNoCorpNum)"
+              @emit-corp-num-validity="corpNumValid = $event"
+            />
+          </v-col>
+          <Designation v-if="showDesignation" cols="12" md="4" />
+        </template>
+
+        <template v-if="isNumberedCompany">
+          <NumberedCompanyBullets />
+        </template>
```
Comment (severinbeauvais) [inline src/components/new-request/search.vue:112]: I suggest moving the v-if into line 111 and getting rid of the template, but this isn't a deal-breaker.

**Review by severinbeauvais — APPROVED** [review]

**Pull request merged**